### PR TITLE
Add design doc for secure plugin installation by CLI

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it
This PR adds the design document for secure plugin installation by CLI
### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
`make lint` passed

### Release note

```release-note
N/A
```

### CODE REVIEW

**Review by vuil — APPROVED** [review]
lgtm

**Comment by anujc25** [comment]
nit can be addressed as follow-up PR.

But we should link this document from some higher level document such as a `docs/full/README.md` file.

**Comment by prkalle** [comment]
> nit: can be addressed as follow-up PR.
> 
> We should link this document from some higher level document such as a `docs/full/README.md` file by adding a security section there.

Thanks @anujc25 . Updated the `docs/full/README.md` with separate section for `secure plugin installation` and explained the UX and also pointed the design doc from README.md

File: docs/full/README.md [hunk 1]
```diff
@@ -228,12 +228,46 @@ reason. Below is a brief summary of these commands
 
 `version`: provides basic version information about the plugin, likely the only common command of broad use to the CLI user.
 
-`info`: provides metadata about the plugin that the CLI will use when presenting information about plugins or when performing lifecycle operations on them.
+`info`: provides metadata about the plugin that the CLI will use when presenting information about plugins or when
```
Comment (vuil) [inline docs/full/README.md:231]: nit: when wrapping, let's wrap consistently to 80 chars for L231-243

File: docs/full/README.md [hunk 2]
```diff
@@ -228,12 +228,46 @@ reason. Below is a brief summary of these commands
...
+## Secure plugin installation
+
+CLI verifies the identity and integrity of the plugin while installing the plugin
+from the repository. You can find more details in
+the [secure plugin installation proposal document](../design/secure-plugin-installation-design.md)
+
+### User experience
+
+CLI verifies cosign signature of the plugin inventory image present in the
+repository. If the signature verification is success, it would download the
```
Comment (vuil) [inline docs/full/README.md:254]: is successful

File: docs/full/README.md [hunk 3]
```diff
@@ -228,12 +228,46 @@ reason. Below is a brief summary of these commands
...
+plugin inventory image on the user's machine and caches to improve the latency
```
Comment (vuil) [inline docs/full/README.md:255]: caches what?

File: docs/full/README.md [hunk 4]
```diff
@@ -228,12 +228,46 @@ reason. Below is a brief summary of these commands
...
+on subsequent plugin installation/search commands.
+If the signature verification fails, CLI would throw an error and stops continuing.
+Signature verification could fail in the below scenarios
```
Comment (vuil) [inline docs/full/README.md:258]: scenarios below:

File: docs/full/README.md [hunk 5]
```diff
@@ -228,12 +228,46 @@ reason. Below is a brief summary of these commands
...
+
+1. Unplanned key rotation: In this case, user either can update to the latest
+   CLI version release with the new key, or users should download the new
+   public key posted in a well known secure location[TBD] to their local file
+   system and export the path of the public key by setting the environment
+   variable `TANZU_CLI_PLUGIN_DISCOVERY_IMAGE_SIGNATURE_PUBLIC_KEY_PATH`.
+2. Repositories without a signature: If users/developers wants to use their own
+   repository without the signature for testing, they can skip the
+   validation(not recommended in production) by appending the repository URL to
```
Comment (vuil) [inline docs/full/README.md:267]: space before (

File: docs/full/README.md [hunk 6]
```diff
@@ -228,12 +228,46 @@ reason. Below is a brief summary of these commands
...
+   the comma-separated list in the environment
+   variable `TANZU_CLI_PLUGIN_DISCOVERY_IMAGE_SIGNATURE_VERIFICATION_SKIP_LIST`.
```
Comment (vuil) [inline docs/full/README.md:269]: An example, e.g. calling "tanzu config set...." would be nice.

**Review by vuil — APPROVED** [review]
Mostly nits but easy to address, feel free to merge once they are.

File: docs/full/README.md (hunk 6 above)
Comment (prkalle) [reply docs/full/README.md:269]: Added example to show how user can add 2 plugin repositories to the signature validation skip list.Thanks

File: docs/full/README.md (hunk 4 above)
Comment (prkalle) [reply docs/full/README.md:258]: updated

File: docs/full/README.md (hunk 3 above)
Comment (prkalle) [reply docs/full/README.md:255]: updated the doc to convey we cache the plugin inventory image.

File: docs/full/README.md (hunk 1 above)
Comment (prkalle) [reply docs/full/README.md:231]: updated.

**Pull request merged**